### PR TITLE
feat: task pagination

### LINKED ISSUE #614: Restore paging on find Tasks api
__Proposal:__
Why do you avoid to return links attribute of the task api response ? Can you restore the possibility to use next link? Or introduce a 
   
 def find_tasks_paged(self, **kwargs):
        """List all tasks.

        :key str name: only returns tasks with the specified name
        :key str after: returns tasks after specified ID
        :key str user: filter tasks to a specific user ID
        :key str org: filter tasks to a specific organization name
        :key str org_id: filter tasks to a specific organization ID
        :key int limit: the number of tasks to return
        :return: Tasks
        """
        return self._service.get_tasks(**kwargs)

with also links sectin in response? 

__Current behavior:__
No link with next task id will be returned because only tasks will be returned 

    def find_tasks(self, **kwargs):
        """List all tasks.

        :key str name: only returns tasks with the specified name
        :key str after: returns tasks after specified ID
        :key str user: filter tasks to a specific user ID
        :key str org: filter tasks to a specific organization name
        :key str org_id: filter tasks to a specific organization ID
        :key int limit: the number of tasks to return
        :return: Tasks
        """
        return self._service.get_tasks(**kwargs).tasks

__Desired behavior:__
Have the next link into the response to implement pagination and  retrieve all tasks (if more than 500) or use smaller chunk during the inquiry 

__Use case:__
Because no pagination method be possible

### PR DESCRIPTION
**Pull request by Sciator**
Closes #614

## Proposed Changes

added `tasks_api.find_tasks_iter`

## Checklist

- [x] CHANGELOG.md updated
- [x] Rebased/mergeable
- [x] A test has been added if appropriate
- [x] `pytest tests` completes successfully
- [x] Commit messages are [conventional](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Sign [CLA](https://www.influxdata.com/legal/cla/) (if not already signed)

### CODE REVIEW

**Comment by Sciator** [comment]
Currently creates one more request than needed. Better identification of last page needed

File: influxdb_client/client/tasks_api.py [hunk 1]
```diff
@@ -37,6 +54,46 @@ def find_tasks(self, **kwargs):
...
+        has_next = tasks_response.links.next is not None
+        last_id = tasks[-1].id if tasks else None
+
+        def next():
+            if has_next and last_id is not None:
+                return self._find_tasks_paged(**{**kwargs, 'after': last_id})
+            else:
+                return [], None
+
+        return tasks, next
```
Comment (bednar) [inline influxdb_client/client/tasks_api.py:80]: This is private function, please move it at bottom of the file.

File: influxdb_client/client/tasks_api.py [hunk 2]
```diff
@@ -37,6 +54,46 @@ def find_tasks(self, **kwargs):
...
+        :key int limit: the number of tasks to return in one page
+        :return: Tasks, Next or None
+        """
+        tasks_response = self._service.get_tasks(**kwargs)
+        tasks = tasks_response.tasks
+
+        has_next = tasks_response.links.next is not None
+        last_id = tasks[-1].id if tasks else None
+
+        def next():
```
Comment (bednar) [inline influxdb_client/client/tasks_api.py:74]: The `next` is build-in Python name, please rename to something like: `next_`.

File: influxdb_client/client/tasks_api.py [hunk 3]
```diff
@@ -37,6 +54,46 @@ def find_tasks(self, **kwargs):
...
+
+        :key str name: only returns tasks with the specified name
+        :key str after: returns tasks after specified ID
+        :key str user: filter tasks to a specific user ID
+        :key str org: filter tasks to a specific organization name
+        :key str org_id: filter tasks to a specific organization ID
+        :key int limit: the number of tasks in one page
+        :return: Tasks iterator
+        """
+        tasks, next = self._find_tasks_paged(**kwargs)
```
Comment (bednar) [inline influxdb_client/client/tasks_api.py:93]: The `next` is build-in Python name, please rename to something like: `next_`.

File: influxdb_client/client/tasks_api.py [hunk 4]
```diff
@@ -11,6 +11,23 @@
     AddResourceMemberRequestBody, RunManually, Run, LogEvent
 
 
+class _TasksIterator:
+    def __init__(self, values, next) -> None:
+        self.values = values
+        self.next = next
```
Comment (bednar) [inline influxdb_client/client/tasks_api.py:17]: The `next` is build-in Python name, please rename to something like: `next_`.

File: influxdb_client/client/tasks_api.py [hunk 5]
```diff
@@ -11,6 +11,23 @@
...
+    def __iter__(self):
+        return self if self.values else (_ for _ in ())
+
+    def __next__(self):
+        if not self.values:
+            if self.next:
+                self.values, self.next = self.next()
+            if not self.values:
+                raise StopIteration
+        return self.values.pop(0)
```
Comment (bednar) [inline influxdb_client/client/tasks_api.py:28]: Please refactor the iterator to a generic format. We should be able to reuse `_TasksIterator` (as `_NextIterator`) in other APIs, such as `organizations_api`.

**Review by bednar — CHANGES_REQUESTED** [review]
Thank you for your PR! 👍 Before we can accept it, there are a few requirements that need to be met:

**Review by bednar — APPROVED** [review]
LGTM :rocket:

**Pull request merged**